tests/unit/test_schema.py

```diff
@@ -69,6 +69,79 @@ def test_schema_is_subset_fails_if_not_subset(module_under_test):
     assert not module_under_test.schema_is_subset(table_schema, tested_schema)
 
 
+@pytest.mark.parametrize(
+    "original_fields,dataframe_fields,expected_difference",
+    [
+        (
+            [
+                {"name": "A", "type": "FLOAT"},
+                {"name": "B", "type": "FLOAT64"},
+                {"name": "C", "type": "STRING"},
+            ],
+            [
+                {"name": "A", "type": "FLOAT64"},
+                {"name": "B", "type": "FLOAT"},
+            ],
+            "Field 'C': no such field in the dataframe.",
+        ),
+        (
+            [
+                {"name": "A", "type": "FLOAT"},
+                {"name": "B", "type": "STRING"},
+            ],
+            [
+                {"name": "A", "type": "FLOAT64"},
+                {"name": "B", "type": "FLOAT"},
+            ],
+            "Field 'B' has different types: dataframe 'FLOAT', BigQuery 'STRING'.",
+        ),
+        (
+            [
+                {"name": "A", "type": "FLOAT"},
+                {"name": "B", "type": "STRING"},
+                {"name": "C", "type": "STRING"},
+            ],
+            [
+                {"name": "A", "type": "FLOAT64"},
+                {"name": "B", "type": "FLOAT"},
+            ],
+            (
+                "Field 'B' has different types: dataframe 'FLOAT', BigQuery 'STRING'.\n"
+                "Field 'C': no such field in the dataframe."
+            ),
+        ),
+        (
+            [
+                {"name": "A", "type": "FLOAT"},
```

Comment:
I think we'll want some tests where the order of the columns doesn't line up. That way we can be sure we aren't showing errors that aren't actually errors.